contracts/modules/cover/Cover.sol

```diff
@@ -781,20 +807,26 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon, ReentrancyGuard {
     return uint(activeCover[assetId].totalActiveCoverInAsset);
   }
 
-  function isPoolAllowed(uint productId, uint poolId) external view returns (bool) {
+  // Returns true if the product exists and the pool is authorized to have the product
+  function isPoolAllowed(uint productId, uint poolId) public view returns (bool) {
 
-    uint poolCount = allowedPools[productId].length;
-
-    if (poolCount == 0) {
-      return true;
-    }
+    // If product exists and is not deprecated, check if the pool is allowed
+    if (doesProductExist(productId) && !isProductDeprecated(productId)) {
```

Comment:
I think we should just inline

return _products[productId].isDeprecated

Initially i thought you're removing this out of isPoolAllowed then it doesn't make sen se to be a separate function mainly cause it's only used here.

My bad if i didn't explain this clearly - if you choose to keep those checks here which i think is fine lets inline the code since those internal functions are not used somewhere else